### ISSUE TITLE
Misleading unreachable code warning from return in try blocks

### ISSUE THREAD

**Issue by scottmcm** [rust-lang/rust#54165]
```rust
#![feature(try_blocks)]

fn first_some() -> Option<()> {
    let _: Option<_> = try { return Some(foo()?) };
    let _: Option<_> = try { return Some(bar()?) };
    None
}
```
https://play.rust-lang.org/?gist=363157c898e0d067260e711db131b256&version=nightly&edition=2018
```
warning: unreachable expression
 --> src/main.rs:4:30
  |
4 |     let _: Option<_> = try { return Some(foo()?) };
  |                              ^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(unreachable_code)] on by default

warning: unreachable expression
 --> src/main.rs:5:30
  |
5 |     let _: Option<_> = try { return Some(bar()?) };
  |                              ^^^^^^^^^^^^^^^^^^^
```
The desugar should probably just suppress this warning for the code it generates.
